patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -6178,6 +6301,26 @@
  		for (int i = 0; i < 1000; i++) {
  			Projectile projectile = Main.projectile[i];
  			if (projectile.active && projectile.owner == Main.myPlayer && ProjectileID.Sets.TurretFeature[projectile.type])
+@@ -38475,6 +_,11 @@
+ 			float B = 0.8f;
+ 			int num = 0;
+ 			int num2 = BiomeTorchHoldStyle(sItem.type);
++			if (num2 >= ItemID.Count) {
++				// This could be a different hook, but then modders would have to move code. TorchID could be used, but it would be a lot of work managing another modded ID and would limit torch lighting and dust behavior to the vanilla pattern.
++				ContentSamples.ItemsByType[num2]?.ModItem?.HoldItem(this);
```

Comment:
Please use `ItemLoader.GetItem(num2)`, it's what we use for other type-based hooks (like `AnglerQuestChat` or `ExtractinatorUse`. It's safer than accidentally modifying the content samples item.

Need to add a comment to `ExampleTorch` when overriding the hook. Note that due to biome select torch god's favour, the player may not actually have an `ExampleTorch` in their inventory when this hook is called, so no modifications should be made to the item instance.

Reply:
Got it.